src/main/java/com/mule/einstein/internal/helpers/PayloadHelper.java

```diff
@@ -107,78 +116,78 @@ public static String executeTools(String originalPrompt, String prompt, String f
       response = getAttributes(findURL.get(0), filePath, extractPayload(ePayload));
       String finalPayload = constructJsonPayload("data: " + response + ", question: " + originalPrompt, paramDetails.getLocale(),
                                                  paramDetails.getProbability());
-      response = executeEinsteinRequest(accessToken, finalPayload, paramDetails.getModelApiName(), URI_MODELS_API_GENERATIONS);
+      response = executeEinsteinRequest(accessTokeDTO, finalPayload, paramDetails.getModelApiName(), URI_MODELS_API_GENERATIONS);
 
     }
     return response;
   }
 
-  private static String executeEinsteinRequest(String accessToken, String payload, String modelName, String resource) {
-    String urlString = URL_BASE + modelName + resource;
-    return executeREST(accessToken, payload, urlString);
+  private static String executeEinsteinRequest(OAuthResponseDTO accessTokenDTO, String payload, String modelName,
+                                               String resource) {
+    String urlString = accessTokenDTO.getApiInstanceUrl() + URI_MODELS_API + modelName + resource;
```

Comment:
We should log this url, this may help for debugging purpose

Reply:
Done